### PR TITLE
Upgrade to Django 2.2

### PR DESCRIPTION
**Pull request by millerdev**
Test failures are expected. Review by commit :blowfish:

### CODE REVIEW

File: corehq/form_processor/models.py [hunk 1]
```diff
@@ -1395,6 +1395,9 @@ def __eq__(self, other):
             self.form_id == other.form_id
         )
 
+    def __hash__(self):
+        return hash((self.case_id, self.type, self.form_id))
```
Comment (esoergel) [inline corehq/form_processor/models.py:1399]: huh, can there not be multiple updates to a case in a single form?  This `__hash__` definition is consistent with `__eq__`, so it seems fine, but I didn't realize that.

**Review by esoergel — COMMENTED** [review]
There's a build failure I haven't looked at, but the changes up to here look good.

**Comment by millerdev** [comment]
@esoergel wrote:
> There's a build failure I haven't looked at

Yeah, sorry the PR description wasn't very informative. I'm using this PR to find and fix test failures caused by the upgrade.

**Comment by dannyroberts** [comment]
Daniel and I offlined and I'm getting us set up to QA this even before all tests pass.

@millerdev there was an issue deploying to staging, which I was also able to repro locally on this branch:

```
$ ./manage.py compilejsi18n
Traceback (most recent call last):
  File "./manage.py", line 189, in <module>
    main()
  File "./manage.py", line 41, in main
    execute_from_command_line(sys.argv)
  File "/Users/danielroberts/.virtualenvs/cchq/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/Users/danielroberts/.virtualenvs/cchq/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/danielroberts/.virtualenvs/cchq/lib/python3.7/site-packages/django/core/management/__init__.py", line 224, in fetch_command
    klass = load_command_class(app_name, subcommand)
  File "/Users/danielroberts/.virtualenvs/cchq/lib/python3.7/site-packages/django/core/management/__init__.py", line 36, in load_command_class
    module = import_module('%s.management.commands.%s' % (app_name, name))
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/danielroberts/.virtualenvs/cchq/lib/python3.7/site-packages/statici18n/management/commands/compilejsi18n.py", line 11, in <module>
    from django.views.i18n import (get_javascript_catalog,
ImportError: cannot import name 'get_javascript_catalog' from 'django.views.i18n' (/Users/danielroberts/.virtualenvs/cchq/lib/python3.7/site-packages/django/views/i18n.py)
```

**Comment by millerdev** [comment]
This PR cannot be merged because of https://github.com/dimagi/commcare-hq/pull/27911/commits/f3b26677a88a01dc0240a752e063cfbde87fc072. All other commits have been cherry-picked into separate PRs, some of which are backward compatible and have been merged, and one that contains non-backward-compatible changes.

The branch has not been deleted yet because the ICDS team is still basing some work on it.

**Pull request closed without merging**